backend/grant/task/models.py

```diff
@@ -0,0 +1,58 @@
+import json
+
+from grant.extensions import ma, db
+from sqlalchemy.ext import mutable
+
+
+class JsonEncodedDict(db.TypeDecorator):
+    """Enables JSON storage by encoding and decoding on the fly."""
+    impl = db.Text
+
+    def process_bind_param(self, value, dialect):
+        if value is None:
+            return '{}'
+        else:
+            return json.dumps(value)
+
+    def process_result_value(self, value, dialect):
+        if value is None:
+            return {}
+        else:
+            return json.loads(value)
+
+
+mutable.MutableDict.associate_with(JsonEncodedDict)
+
+
+class Task(db.Model):
+    __tablename__ = 'task'
+
+    id = db.Column(db.Integer(), primary_key=True)
+    job_id = db.Column(db.Integer(), nullable=False)
```

Comment:
Nit: I think of IDs as being unique identifiers. This seems to me more of a `job_type` or something like that.

Reply:
agreed!